tools/clang/test/CodeGenSPIRV_Lit/spv.inline.builtin.output.hlsl

```diff
@@ -0,0 +1,16 @@
+// RUN: %dxc -T ps_6_0 -E main -fcgl  %s -spirv | FileCheck %s
+
+// CHECK: OpEntryPoint Fragment %main "main" [[fragStencilVar:%[0-9]+]]
+// CHECK: OpDecorate [[fragStencilVar]] BuiltIn FragStencilRefEXT
+
+// CHECK: [[fragStencilVar]] = OpVariable %_ptr_Output_int Output
+
+//[[vk::ext_extension("SPV_EXT_shader_stencil_export")]]
+[[vk::ext_builtin_output(/* FragStencilRefEXT */ 5014)]]
```

Comment:
Just by curiosity I tried this:

```
  9 [[vk::ext_builtin_input(/* FragStencilRefEXT */ 5014)]]
 10 [[vk::ext_builtin_output(/* FragStencilRefEXT */ 5014)]]
 11 static int gl_FragStencilRefARB;
```

Because the input condition is first, the input decoration is chosen. But there are no diagnostic to say I'm doing something wrong.
Also, if I set it to input, I get a validation error as the input with a type float or int must be marked Flat.

```
fatal error: generated SPIR-V is invalid: [VUID-StandaloneSpirv-Flat-04744] Fragment OpEntryPoint operand 2 with Input interfaces with integer or float type must have a Flat decoration for Entry Point id 1.
  %2 = OpVariable %_ptr_Input_int Input

note: please file a bug report on https://github.com/Microsoft/DirectXShaderCompiler/issues with source code if possible
```

(To make this trigger, I replaced the line `gl_FragStencilRefARB = 10;` with `gl_FragStencilRefARB;`

Reply:
I've created an error diagnostic for the first case, and I've handled the second automatically adding the `Flat` decoration in the necessary cases.